mirdata/datasets/cante100.py

```diff
@@ -417,13 +339,72 @@ def __init__(self, data_home=None):
             data_home,
             index=DATA.index,
             name="cante100",
-            track_object=Track,
+            track_class=Track,
             bibtex=BIBTEX,
             remotes=REMOTES,
             download_info=DOWNLOAD_INFO,
             license_info=LICENSE_INFO,
         )
 
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(self.data_home, "cante100Meta.xml")
+        if not os.path.exists(metadata_path):
+            logging.info(
+                "Metadata file {} not found.".format(metadata_path)
+                + "You can download the metadata file for cante100 "
+                + "by running cante100.download()"
+            )
+            return None
+
+        tree = ET.parse(metadata_path)
+        root = tree.getroot()
+
+        # ids
+        indexes = []
+        for child in root:
+            index = child.attrib.get("id")
+            if len(index) == 1:
+                index = "00" + index
+                indexes.append(index)
+                continue
+            if len(index) == 2:
+                index = "0" + index
+                indexes.append(index)
+                continue
+            else:
+                indexes.append(index)
+
+        # musicBrainzID
+        identifiers = [ident.text for ident in root.iter("musicBrainzID")]
+
+        # artist
+        artists = [artist.text for artist in root.iter("artist")]
+
+        # titles
+        titles = [title.text for title in root.iter("title")]
+
+        # releases
+        releases = [release.text for release in root.iter("anthology")]
+
+        # duration
+        minutes = [float(minute.text) * 60 for minute in root.iter("duration_m")]
+        seconds = [float(second.text) for second in root.iter("duration_s")]
+        durations = [m + s for (m, s) in zip(minutes, seconds)]
+
+        metadata = dict()
+        metadata["data_home"] = self.data_home
+        for i, j in zip(indexes, range(len(artists))):
```

Comment:
This comment probably doesn't belong here, but at this point I'm understanding that for track-level metadata we’re assuming that the metadata at dataset-level is a dictionary? And this is not enforced in the code but left to the contributor right? i.e. the core functionality checks if the `track_id` is in `metadata` so the contributor should write the code accordingly

Reply:
So is some sort of metadata index?

Reply:
it's admittedly a bit hacky - 90% of the metadata files are dictionaries mapping track ids to metadata, but some are not (e.g. ikala, which is keyed by "singer id"), which is why we check if the track id is a key or not.

In cante it falls into the "regular" case, where the key is the track id.